near-sdk/src/promise.rs

```diff
@@ -464,8 +464,8 @@ impl Promise {
     /// p1.then(p2).and(p3).then(p4);
     /// ```
     /// Uses low-level [`crate::env::promise_batch_then`]
-    pub fn then(self, mut other: Promise) -> Promise {
-        match &mut other.subtype {
+    pub fn then(self, other: Promise) -> Promise {
+        match &other.subtype {
```

Comment:
Note: This is no-op change without any implications for the function signature. Just a unnecessary `mut`, which I removed to keep it consistent between `then` and `then_concurrent`.